src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/client/NonAssociationsApiClient.kt

```diff
@@ -0,0 +1,43 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.client
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.core.ParameterizedTypeReference
+import org.springframework.stereotype.Component
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.core.publisher.Mono
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.prison.api.OffenderNonAssociationDetailsDto
+import java.time.Duration
+
+@Component
+class NonAssociationsApiClient(
+  @Qualifier("nonAssociationsApiWebClient") private val webClient: WebClient,
+  @Value("\${prison.api.timeout:60s}") private val apiTimeout: Duration,
+) {
+
+  companion object {
+    val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+    private val TYPE_FOR_OFFENDER_NONASSOCIATION = object : ParameterizedTypeReference<OffenderNonAssociationDetailsDto>() {}
+  }
+
+  fun getOffenderNonAssociation(offenderNo: String): OffenderNonAssociationDetailsDto? {
+    LOG.debug("Entered getOffenderNonAssociation $offenderNo")
+    return webClient.get()
+      .uri("/legacy/api/offenders/$offenderNo/non-association-details?currentPrisonOnly=false&excludeInactive=true")
```

Comment:
assume you want other NA's in other prisons - as your code doesn't filter - BUT not sure why you consider prisoners outside current prison (and released ones)

Reply:
Not sure why this was originally done this way, before I joined the team. The only assumption I can come up with is when they move prison and the NA is not updated, but seems unlikely,.. If you are absolutely sure there is no reason to consider prisoners outside the current prison then please change.

Reply:
ok will do